Add scripts and a cron to export/import RabbitMQ definitions

bdero opened the pull request:
This PR adds a script and cron for exporting RabbitMQ definitions to a given location. It also adds a script for importing RabbitMQ definitions restored from backup. CC @smarnach

bdero committed:
Add scripts and a cron to export/import RabbitMQ definitions

bdero commented:
This change goes along with https://github.com/open-craft/deployment-deploy-services/pull/16

smarnach commented on defaults/main.yml:
```diff
@@ -13,3 +13,9 @@ RABBITMQ_KEY_FILE: "/etc/letsencrypt/live/{{ RABBITMQ_HOSTNAME }}/privkey.pem"
   - python-software-properties
   - python-letsencrypt
+
+RABBITMQ_EXPORT_DIR: /usr/local/lib/rabbitmq/backup
```
This should be in `/var`, not in `/usr`.  (`/usr` should only be changed when installing new software.)

bdero replied:
I changed this to `/var/local/rabbitmq/backup`. Does that seem like a reasonable path?

smarnach replied:
I think if we'd follow the FHS strictly, this would have to go in `/var/cache`, but `/var/local` is fine with me.

smarnach commented on files/rabbitmq-backup:
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+DEST="$1"
+python /var/lib/rabbitmq/mnesia/*plugins-expand/rabbitmq_management-*/priv/www/cli/rabbitmqadmin export $DEST -c /etc/rabbitmqadmin.conf -N host_ssl -q
```
This is a rather weird place for the admin script.  The [documented way to get the script](https://www.rabbitmq.com/management-cli.html) seems to be to download it from https://rabbitmq.net.opencraft.hosting:15671/cli/, so maybe we should do that in the Ansible playbook and store it in `/usr/local/sbin`.  However, I can't access this URL at the moment (the connection times out when I try).

bdero replied:
Yeah, I wanted to avoid having to fetch it, but in the end it's more weird because of the globs to avoid version numbers. I changed it to fetch the command from https://rabbitmq.net.opencraft.hosting:15671/cli/. Thanks!

smarnach replied:
Thanks.  I mostly prefer the new version because that's the documented place to get the script, whereas the path you used before seems to be an implementation detail we should not rely on.

smarnach commented on tasks/main.yml:
```diff
@@ -60,6 +60,41 @@
+    group: root
+    mode: 700
+
+- name: set up cron job for rabbitmq exporting
```
We don't need a separate cron job here.  It's easier to use [`TARSNAP_BACKUP_PRE_SCRIPT`](https://github.com/open-craft/ansible-backup-to-tarsnap/blob/45ac5b4bfd9fe4003c126c3284bb0a91ac48b86a/defaults/main.yml#L19) instead.

bdero replied:
Ah, I didn't think to look for this. Thanks!

smarnach commented on defaults/main.yml:
```diff
@@ -13,3 +13,9 @@ RABBITMQ_KEY_FILE: "/etc/letsencrypt/live/{{ RABBITMQ_HOSTNAME }}/privkey.pem"
   - python-letsencrypt
+
+RABBITMQ_EXPORT_DIR: /usr/local/lib/rabbitmq/backup
+RABBITMQ_SCRIPTS_DIR: /usr/local/bin
```
This should be `/usr/local/sbin`, since you set the permissions to only let root execute it.

bdero replied:
Done, thanks.

smarnach reviewed (changes requested):
@bdero Thanks for taking this on!  I've added a few comments.

bdero committed:
Fetch the rabbitmqadmin command from the management webserver

bdero commented:
@smarnach Thanks for the thorough review! I've addressed your comments in the latest commit on this PR and https://github.com/open-craft/deployment-deploy-services/pull/16.

smarnach commented:
Looks all good to me now, :+1:.
